### PR TITLE
Python API Support for Fixed Source Calculations

### PR DESCRIPTION
**Pull request by wbinventor**
This PR extends the Python API to support fixed source calculations. As pointed out by @paulromano in #506, the `Settings` class did not include support for the `<fixed_source>` XML attribute. This PR adds a new `Settings._run_mode` class attribute which may be set to either "eigenvalue" (default) or "fixed source". This attribute is used to determine whether or not to create an `<eigenvalue>` or `<fixed_source>` block in the XML input when `Settings.export_to_xml()` is called.

### CODE REVIEW

File: openmc/settings.py [hunk 1]
```diff
@@ -399,6 +407,14 @@ def dd_allow_leakage(self):
     def dd_count_interactions(self):
         return self._dd_count_interactions
 
+    @run_mode.setter
+    def run_mode(self, run_mode):
+        if not 'run_mode' in ['eigenvalue', 'fixed source']:
```
Comment (paulromano) [inline openmc/settings.py]: The more readable `a not in b` is preferred over `not a in b`

Comment (wbinventor) [inline openmc/settings.py]: Old habits die hard. I just pushed the more Pythonic version.

**Comment by paulromano** [comment]
Thanks for the quick fix @wbinventor!

**Comment by juankost** [comment]
I believe the fixed source calculation still isn't working (at least to me it isn't).
I think the problem is :
 def run_mode(self, **run_mode**):
 -        if not 'run_mode' in ['eigenvalue', 'fixed source']:
 +        if **'run_mode'** not in ['eigenvalue', 'fixed source']:

Instead of checking if the variable **run_mode** is in  ['eigenvalue', 'fixed source'], we are checking if the string **'run_mode'** is in  ['eigenvalue', 'fixed source'],  (which obviously isn't).

**Comment by smharper** [comment]
@juankost, that error has been fixed on our develop branch with a5c6a940991bcf5b99fcb1dad98c5b47d7879708

**Pull request merged**